pkg/cmd/cli/uninstall/uninstall.go

```diff
@@ -348,23 +354,19 @@ func gracefullyDeleteResource(ctx context.Context, kbClient kbclient.Client, nam
 func waitDeletingResources(ctx context.Context, kbClient kbclient.Client, namespace string) error {
 	// Wait for the deletion of all the restores within a specified time frame
 	err := wait.PollImmediate(time.Second, gracefulDeletionMaximumDuration, func() (bool, error) {
-		restoreList := &velerov1api.RestoreList{}
-		if errList := kbClient.List(ctx, restoreList, &kbclient.ListOptions{Namespace: namespace}); errList != nil {
-			return false, errList
-		}
-
-		dataUploadList := &velerov2alpha1api.DataUploadList{}
-		if errList := kbClient.List(ctx, dataUploadList, &kbclient.ListOptions{Namespace: namespace}); errList != nil {
-			return false, errList
-		}
-
-		dataDownloadList := &velerov2alpha1api.DataDownloadList{}
-		if errList := kbClient.List(ctx, dataDownloadList, &kbclient.ListOptions{Namespace: namespace}); errList != nil {
-			return false, errList
+		itemLen := 0
+		for i := range resToDelete {
+			if errList := kbClient.List(ctx, resToDelete[i], &kbclient.ListOptions{Namespace: namespace}); errList != nil {
+				return false, errList
+			}
+			itemLen += reflect.ValueOf(resToDelete[i]).Elem().FieldByName("Items").Len()
+			if itemLen > 0 {
+				fmt.Print(".")
+				return false, nil
+			}
 		}
 
-		if len(restoreList.Items)+len(dataUploadList.Items)+len(dataDownloadList.Items) > 0 {
-			fmt.Print(".")
+		if itemLen > 0 {
```

Comment:
Seems this line is useless as the function always returns in line 365 if itemLen > 0?

Reply:
Indeed it's useless, it's But If I do not add `itemLen > 0 `, the compile will failed with a `missing return`

Reply:
Why? You remove both code blocks of `if` and `else`? Shouldn't you just remove the `if` block and keep the last `return`?

Reply:
it's reported by ConditionFunc in PollImmediate, ConditionFunc should have a return for every condition